src/features/Authentication/organisms/SignUpForm/SignUpForm.js

```diff
@@ -0,0 +1,129 @@
+}
+
+const SignUpForm = () => {
+  const [{ fetching, error, data }, createUser] = useMutation(CREATE_USER);
+  const onSubmit = async (values, { setSubmitting }) => {
+    await createUser(values);
+    setSubmitting(false);
+  }
+
+  return (
+    <div>
+      <Formik
+        initialValues={INITIAL_VALUES}
+        validationSchema={SIGNUP_SCHEMA}
+        onSubmit={onSubmit}
+      >
+        {({ isSubmitting }) => (
+          <Grid item xs={12} sm={8} md={5}>
+            <Box
+              sx={{
+                my: 8,
+                mx: 4,
+                display: 'flex',
+                flexDirection: 'column',
+                alignItems: 'center',
+              }}
+            >
+              <Typography variant="h3" component="div" gutterBottom align="center">
+                Sign Up
+              </Typography>
+
+              {(data && data.createUser !== null) &&
+                <Grid container spacing={2} sx={{ mt: 3, mb: 3 }}>
+                  <Grid item xs={12}>
+                    <Alert severity="success" align='left'>
+                      <AlertTitle>Success</AlertTitle>
+                      User was registered
+                    </Alert>
+                  </Grid>
+                </Grid>
+              }
+              {error &&
+                <Grid container spacing={2} sx={{ mt: 3, mb: 3 }}>
+                  <Grid item xs={12}>
+                    <Alert severity="error" align='left'>
+                      <AlertTitle>Error</AlertTitle>
+                      {error.message}
+                    </Alert>
+                  </Grid>
+                </Grid>
+              }
+              <Form>
+                <Grid container spacing={2}>
+                  <Grid item xs={12}>
+                    <Field component={TextField} name="firstName" fullWidth label="First name" placeholder="First name" variant="outlined" />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Field component={TextField} name="lastName" fullWidth label="Last name" placeholder="Last name" variant="outlined" />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Field component={TextField} name="email" type="email" fullWidth label="Email" placeholder="email@mail.com" variant="outlined" />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Field component={TextField} name="password" type="password" fullWidth label="Password" placeholder="Password" variant="outlined" />
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Button
+                      aria-label="submit"
+                      type="submit"
+                      fullWidth
+                      variant="contained"
+                      sx={{ mt: 3, mb: 2 }}
+                      disabled={isSubmitting}
+                    >
+                      {fetching ? <CircularProgress color="inherit" /> : "Sign Up"}
```

Comment:
Can be moved inside the button conponent

Reply:
@VladWithCookies what's do you mean?

Reply:
@AlexNGerman it would be nice to create a new molecule called SubmitButton and manage displaying of spinner inside this component